arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -400,56 +410,59 @@ arangodb::aql::AqlValue aqlFnTokens(arangodb::aql::ExpressionContext* expression
   arangodb::iresearch::IResearchAnalyzerFeature::AnalyzerPool::ptr pool;
 
   if (trx) {
-    auto* sysDatabase = arangodb::application_features::ApplicationServer::lookupFeature< // find feature
-      arangodb::SystemDatabaseFeature // featue type
-    >();
+    auto* sysDatabase = arangodb::application_features::ApplicationServer::lookupFeature<  // find feature
+        arangodb::SystemDatabaseFeature  // featue type
+        >();
 
     auto sysVocbase = sysDatabase ? sysDatabase->use() : nullptr;
 
     if (sysVocbase) {
       pool = analyzers->get(name, trx->vocbase(), *sysVocbase);
     }
   } else {
-    pool = analyzers->get(name); // verbatim
+    pool = analyzers->get(name);  // verbatim
   }
 
   if (!pool) {
-    auto const message = "failure to find arangosearch analyzer with name '"s
-      + static_cast<std::string>(name)
-      + "' while computing result for function 'TOKENS'";
+    auto const message = "failure to find arangosearch analyzer with name '"s +
+                         static_cast<std::string>(name) +
+                         "' while computing result for function 'TOKENS'";
 
     LOG_TOPIC("0d256", WARN, arangodb::iresearch::TOPIC) << message;
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER, message);
   }
 
-  auto analyzer = pool->get();
+  auto string_analyzer = pool->get();
 
-  if (!analyzer) {
-    auto const message = "failure to find arangosearch analyzer with name '"s
-      + static_cast<std::string>(name)
-      + "' while computing result for function 'TOKENS'";
+  if (!string_analyzer) {
+    auto const message = "failure to get arangosearch analyzer with name '"s +
+                         static_cast<std::string>(name) +
+                         "' while computing result for function 'TOKENS'";
     LOG_TOPIC("d7477", WARN, arangodb::iresearch::TOPIC) << message;
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER, message);
   }
 
-  if (!analyzer->reset(data)) {
-    auto const message = "failure to reset arangosearch analyzer: ' "s
-      + static_cast<std::string>(name)
-      + "' while computing result for function 'TOKENS'";
+  auto& string_terms = string_analyzer->attributes().get<irs::term_attribute>();
 
-    LOG_TOPIC("45a2d", WARN, arangodb::iresearch::TOPIC) << message;
+  if (!string_terms) {
+    auto const message =
+        "failure to retrieve values from arangosearch analyzer name '"s +
+        static_cast<std::string>(name) +
+        "' while computing result for function 'TOKENS'";
+
+    LOG_TOPIC("f46f2", WARN, arangodb::iresearch::TOPIC) << message;
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL, message);
   }
 
-  auto& values = analyzer->attributes().get<irs::term_attribute>();
+  irs::numeric_token_stream numeric_analyzer;
```

Comment:
there was an idea to instantiate `numeric_token_stream` lazily as calling `TOKENS(...)` with numeric terms is rare case